services/subscriptions/subscriptions.go

```diff
@@ -0,0 +1,80 @@
+package subscriptions
+
+import (
+	"fmt"
+	"sync"
+	"time"
+)
+
+type Subscriptions struct {
+	mu          sync.Mutex
+	subs        map[SubscriptionID]*Subscription
+	checkPeriod time.Duration
+}
+
+func NewSubscriptions(period time.Duration) *Subscriptions {
+	return &Subscriptions{
+		mu:          sync.Mutex{},
+		subs:        make(map[SubscriptionID]*Subscription),
+		checkPeriod: period,
+	}
+}
+
+func (subs *Subscriptions) Create(namespace string, filter filter) (SubscriptionID, error) {
+	subs.mu.Lock()
+	defer subs.mu.Unlock()
+
+	newSub := NewSubscription(namespace, filter)
+
+	go newSub.Start(subs.checkPeriod)
+
+	subs.subs[newSub.id] = newSub
```

Comment:
Maybe rename `subs` to `s` so we write `s.subs` and there is less chance for mixing the 2 names?